src/component.jl

```diff
@@ -440,10 +440,30 @@ function prepare_for_removal!(component::InfrastructureSystemsType)
     @debug "cleared all forecast data from" component
 end
 
+"""
+Returns an iterator of Forecast instances attached to the component.
+"""
 function iterate_forecasts(component::InfrastructureSystemsType)
     container = get_forecasts(component)
     forecast_keys = sort!(collect(keys(container.data)), by = x -> x.initial_time)
 
+    Channel() do channel
```

Comment:
This is a secondary issue Sourabh saw today.  This needs to return public forecasts, not internal ones.